functional/README.md

```diff
@@ -1,3 +1,63 @@
+## configure environment
+
+### Debian/Ubuntu
+
+#### Install Vagrant
+
+```sh
+sudo apt-get install -y git nfs-kernel-server
+wget https://releases.hashicorp.com/vagrant/1.8.1/vagrant_1.8.1_x86_64.deb
+sudo dpkg -i vagrant_1.8.1_x86_64.deb
+```
+
+#### Install VirtualBox
+
+```sh
+echo "deb http://download.virtualbox.org/virtualbox/debian $(lsb_release -sc) contrib" | sudo tee /etc/apt/sources.list.d/virtualbox.list
+wget -q https://www.virtualbox.org/download/oracle_vbox.asc -O- | sudo apt-key add -
+sudo apt-get update
+sudo apt-get install -y build-essential dkms
+sudo apt-get install -y VirtualBox-5.0
+#Previous VirtualBox (if you have problems with nested virtualization, more info here: https://www.virtualbox.org/ticket/14965)
+#sudo apt-get install -y VirtualBox-4.3
+```
+
+### Centos/Fedora
+
+**NOTE**: NFS doesn't work on CentOS 6.x
+
+#### Install Vagrant
+
+```sh
+sudo yum install -y git nfs-utils
+sudo reboot
+sudo service nfs start
+sudo yum install -y https://releases.hashicorp.com/vagrant/1.8.1/vagrant_1.8.1_x86_64.rpm
+```
+
+#### Install VirtualBox
+
+```sh
+source /etc/os-release
+for id in $ID_LIKE $ID; do break; done
+OS_ID=${id:-rhel}
+curl http://download.virtualbox.org/virtualbox/rpm/$OS_ID/virtualbox.repo | sudo tee /etc/yum.repos.d/virtualbox.repo
+sudo yum install -y make automake gcc gcc-c++ kernel-devel-`uname -r` dkms
+sudo yum install -y VirtualBox-5.0
+#Previous VirtualBox (if you have problems with nested virtualization, more info here: https://www.virtualbox.org/ticket/14965)
+#sudo yum install -y VirtualBox-4.3
+```
+
+## run tests in vagrant
+
+```sh
+git clone https://github.com/coreos/fleet
+cd fleet/functional
+# wget https://raw.githubusercontent.com/coreos/coreos-vagrant/master/Vagrantfile
+# we will use default unmodified CoreOS Vagrantfile
+./run-in-vagrant
+```
+
 ## fleet functional tests
 
 This functional test suite deploys a fleet cluster using nspawn containers, and asserts fleet is functioning properly.
```

Comment:
This intro should still be at the top of the file, otherwise the configure stuff lacks context